packages/@aws-cdk/aws-elasticloadbalancingv2/lib/shared/base-listener.ts

```diff
@@ -28,28 +28,31 @@ export abstract class BaseListener extends Resource {
    * Validate this listener
    */
   protected validate(): string[] {
-    if (this.defaultActions.length === 0) {
-      return ['Listener needs at least one default target group (call addTargetGroups)'];
+    if (!this.defaultAction) {
+      return ['Listener needs at least one default action or target group (call addTargetGroups or addAction)'];
     }
     return [];
   }
 
   /**
-   * Add an action to the list of default actions of this listener
+   * Configure the default action
+   *
    * @internal
    */
-  protected _addDefaultAction(action: CfnListener.ActionProperty) {
-    this.defaultActions.push(action);
-  }
+  protected _setDefaultAction(action: IListenerAction) {
+    // It might make sense to 'throw' here.
+    //
+    // However, programs may already exist out there which configured an action twice,
+    // in which case the second action accidentally overwrite the initial action, and in some
+    // way ended up with a program that did what the author intended. If we were to add throw now,
+    // the previously working program would be broken.
+    //
+    // Instead, signal this through a warning.
+    // @deprecate: upon the next major version bump, replace this with a `throw`
+    if (this.defaultAction) {
+      this.node.addWarning('A default Action already existed on this Listener and was replaced. Configure exactly one default Action.');
+    }
```

Comment:
TBH - I don't like this warning here. No one ever checks this, nor do we collect it in our issue reporting. Has this helped us or the customer debug in the past?

Opportunity to use a feature flag instead?